docs/standard/native-interop/tutorial-comwrappers.md

```diff
@@ -0,0 +1,430 @@
+
+Once the `CreateObject()` method is implemented, the `ComWrappers` subclass will be able to produce Native Object Wrappers for COM instances that implement both `IDemoGetType` and `IDemoStoreType`.
+
+```csharp
+IntPtr iunk = ...; // Get a COM instance from native code.
+object rcw = cw.GetOrCreateObjectForComInstance(iunk, CreateObjectFlags.UniqueInstance);
+```
+
+### Step 4 &ndash; Handle Native Object Wrapper lifetime details
+
+The `ComputeVtables()` and `CreateObject()` implementations covered some wrapper lifetime details, but there are further considerations. While this can be a short step, it can also significantly increase the complexity of the `ComWrappers` design.
+
+Unlike the Managed Object Wrapper, which is controlled by calls to its `AddRef()` and `Release()` methods, the lifetime of a Native Object Wrapper is nondeterministically handled by the GC. The question here is, when does the Native Object Wrapper call `Release()` on the `IntPtr` that represents the COM instance? There are two general buckets:
+
+1) The Native Object Wrapper's Finalizer is responsible for calling the COM instance's `Release()` method. This is the only time when it's safe to call this method. At this point, it's been correctly determined by the GC that there are no other references to the Native Object Wrapper in the .NET runtime. There can be complexity here if you're properly supporting COM Apartments; for more information, see the [Additional considerations](#additional-considerations) section.
+
+2) The Native Object Wrapper implements `IDisposable` and calls `Release()` in `Dispose()`.
+
+  > [!NOTE]
+  > The `IDisposable` pattern should only be supported if, during the `CreateObject()` call, the `CreateObjectFlags.UniqueInstance` flag was passed in. If this requirement is not followed, it's possible for disposed Native Object Wrappers to be reused after being disposed.
+
+## Using the `ComWrappers` subclass
+
+You now have a `ComWrappers` subclass that can be tested. To avoid creating a native library that returns a COM instance that implements `IDemoGetType` and `IDemoStoreType`, you'll use the Managed Object Wrapper and treat it as a COM instance &ndash; this must be possible in order to pass it COM anyways.
+
+Let's create a Managed Object Wrapper first. Instantiate a `DemoImpl` instance and display its current string state.
+
+```csharp
+var demo = new DemoImpl();
+
+string? value = demo.GetString();
+Console.WriteLine($"Initial string: {value ?? "<null>"}");
+```
+
+Now you can create an instance of `DemoComWrappers` and a Managed Object Wrapper that you can then pass into a COM environment.
+
+```csharp
+var cw = new DemoComWrappers();
+
+IntPtr ccw = cw.GetOrCreateComInterfaceForObject(demo, CreateComInterfaceFlags.None);
```

Comment:
I didn't see this sample mention how to handle cases where the code want to `CoCreateInstance` the COM object? I think it might be a relatively common case. Does it need to PInvoke into the Win32 API, or is there some managed helper to make it easier?

Reply:
>I think it might be a relatively common case.

Agreed. I am going to leave that out. It is conceptually simple once you try it. I don't think it is worth adding that concept here though.

Reply:
Sorry - does that mean that I should PInvoke into `CoCreateInstance` then? (I agree that it's probably pretty simple to do so, I'm just curious if there's better way)

Reply:
Yep. Implicit activation has been removed from the runtime but could technically be re-built with `ComWrappers` like C#/WinRT.